src/Microsoft.Android.Sdk.ILLink/TypeMappingStep.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Java.Interop.Tools.Cecil;
+using Java.Interop.Tools.TypeNameMappings;
+using Mono.Cecil;
+using Mono.Linker;
+using Mono.Linker.Steps;
+
+namespace Microsoft.Android.Sdk.ILLink;
+
+/// <summary>
+/// MVP "typemap" implementation for NativeAOT
+/// </summary>
+public class TypeMappingStep : BaseStep
+{
+	const string AssemblyName = "Microsoft.Android.Runtime.NativeAOT";
+	const string TypeName = "Android.Runtime.NativeAOT.NativeAotTypeManager";
+	readonly IDictionary<string, TypeDefinition> TypeMappings = new Dictionary<string, TypeDefinition> (StringComparer.Ordinal);
+	AssemblyDefinition? MicrosoftAndroidRuntimeNativeAot;
+
+	protected override void ProcessAssembly (AssemblyDefinition assembly)
+	{
+		if (assembly.Name.Name == AssemblyName) {
+			MicrosoftAndroidRuntimeNativeAot = assembly;
+			return;
+		}
+		if (Annotations?.GetAction (assembly) == AssemblyAction.Delete)
+			return;
+		if (!assembly.MainModule.HasTypeReference ("Java.Lang.Object"))
+			return;
+
+		foreach (var type in assembly.MainModule.Types) {
+			ProcessType (assembly, type);
+		}
+	}
+
+	protected override void EndProcess ()
+	{
+		if (MicrosoftAndroidRuntimeNativeAot is null) {
+			Context.LogMessage ($"Unable to find {AssemblyName} assembly");
+			return;
```

Comment:
We are getting here, so will keep working on it this afternoon.

It doesn't appear the assembly is preserved, but I thought UCO would preserve it:

https://github.com/dotnet/android/blob/cc63af6f8d3b1147fcddaaac5b51459b3e1caf23/src/Microsoft.Android.Runtime.NativeAOT/Android.Runtime.NativeAOT/JavaInteropRuntime.cs#L32-L34

Reply:
The solution to this problem was:
```xml
  <ItemGroup>
    <UnmanagedEntryPointsAssembly Include="Microsoft.Android.Runtime.NativeAOT" />
  </ItemGroup>
```
So, will put this in our targets by default.